Resources/config/services.xml

```diff
@@ -1,61 +1,63 @@
 <?xml version="1.0" ?>
 
 <container xmlns="http://symfony.com/schema/dic/services"
-           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
 
     <services>
-        <service id="Shivas\VersioningBundle\Formatter\FormatterInterface"
-                 alias="Shivas\VersioningBundle\Formatter\GitDescribeFormatter"/>
-        <service id="Shivas\VersioningBundle\Writer\WriterInterface"
-                 alias="Shivas\VersioningBundle\Writer\VersionWriter"/>
-
-        <service id="Shivas\VersioningBundle\Service\VersionManager">
-            <argument type="service" id="Symfony\Component\Cache\Adapter\AdapterInterface"/>
+        <service id="Shivas\VersioningBundle\Service\VersionManager" public="true">
+            <argument type="service" id="shivas_versioning.cache.version"/>
             <argument type="service" id="Shivas\VersioningBundle\Writer\WriterInterface"/>
             <argument type="service" id="Shivas\VersioningBundle\Formatter\FormatterInterface"/>
         </service>
 
-        <service id="Shivas\VersioningBundle\Formatter\GitDescribeFormatter"/>
-        <service id="Shivas\VersioningBundle\Writer\VersionWriter">
+        <service id="Shivas\VersioningBundle\Formatter\FormatterInterface" alias="shivas_versioning.formatter.git" public="false"/>
+        <service id="Shivas\VersioningBundle\Writer\WriterInterface" alias="shivas_versioning.writer.version" public="false"/>
+
+        <service id="shivas_versioning.formatter.git" class="Shivas\VersioningBundle\Formatter\GitDescribeFormatter" public="false"/>
+        <service id="shivas_versioning.writer.version" class="Shivas\VersioningBundle\Writer\VersionWriter" public="false">
             <argument>%kernel.project_dir%</argument>
         </service>
 
-        <service id="Shivas\VersioningBundle\Provider\VersionProvider">
+        <service id="shivas_versioning.provider.version" class="Shivas\VersioningBundle\Provider\VersionProvider" public="false">
             <argument>%kernel.project_dir%</argument>
             <tag name="shivas_versioning.provider" alias="version" priority="0"/>
```

Comment:
I think this provider should have a higher priority, to stay before providers registered without a priority, as it is meant to win over anything else to make manual versioning working.

Reply:
Good catch, that was indeed the intention! Thanks for the useful feedback!